lib/pbench/server/api/__init__.py

```diff
@@ -63,361 +60,24 @@ def create_app():
         sys.exit(1)
 
     try:
-        config = PbenchServerConfig(cfg_name)
+        return PbenchServerConfig(cfg_name)
     except BadConfig as e:
         print(f"{__name__}: {e} (config file {cfg_name})", file=sys.stderr)
         sys.exit(1)
 
+
+def create_app():
+    """Create Flask app with defined resource endpoints."""
+
     app = Flask(__name__)
```

Comment:
Is `__name__` going to resolve to "`api`" or to "`__init__`"?  If the former, great; if the latter, then let's not use `__name__` but perhaps `api-server`, or something like that.

Reply:
I am not 100% sure but I think it will resolve to the module name. ref: https://flask.palletsprojects.com/en/1.1.x/tutorial/factory/ so in our case `pbench.server.api`

> `__name__` is the name of the current Python module. The app needs to know where it’s located to set up some paths, and `__name__` is a convenient way to tell it that.

Reply:
Try this out:

```
$ cat this.py
class Foo:
    def __init__(self):
        self.myname = "foo"

    def bar(self):
        print(self.myname, __name__)


def this():
    print(__name__)

print(__name__)
this()
Foo().bar()
$ python3 this.py
__main__
__main__
foo __main__
```

Reply:
the yes but we dont directly execute the `__init__` file, we execute the file via gunicorn. to do that we import `create_app` in `shell.py` file `from pbench.server.api import create_app` and if I execute shell.py file I get `pbench.server.api` when I do `print(__name__)` inside `create_app()`

Reply:
Right, so that is the module name, but we want a logger for the instance of the service.  I am not sure we should name the logger based on the module.  At least, in other parts of the server code, we name the logger explicitly, see https://github.com/distributed-system-analysis/pbench/blob/master/lib/pbench/server/indexer.py#L4415.

Reply:
Ahh I see, thanks for the clarification